### PR TITLE
FES throws an error when non globally exposed function's name used in other context

### LINKED ISSUE #5753: Version 1.4.1 adds reserved function “size”
### Topic

Between version 1.4.0 and 1.4.1 a reserved function “size” was added. I feel like this variable name is used way too commonly in sketches to have it as a reserved function.

### PR DESCRIPTION
**Pull request by aditya-shrivastavv**
### Changes:
Previously we were adding all the methods of all p5 classes as reserved functions in the `functionArray`, but all p5 classes do not have their functions added to the global scope.
So, as suggested by @davepagurek that only the methods/functions inside `p5`, `p5.Renderer`, `p5.RendererGL`, and `p5.Renderer2D` are available in the global scope, so only those should be added to `functionArray` so that FES do not throw an error when a function's name (which is not exposed or defined by p5.js in the global scope) is used as a variable or any other way.

#### PR Checklist

- [x] `npm run lint` passes
- [ ] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

File: src/core/friendly_errors/sketch_reader.js [hunk 1]
```diff
@@ -319,12 +319,14 @@ if (typeof IS_MINIFIED !== 'undefined') {
       }
     }
     const keyArray = Object.keys(p5Constructors);
+    const globalFunctions = ['Renderer', 'Renderer2D', 'RendererGL'];
```
Comment (aditya-shrivastavv) [inline src/core/friendly_errors/sketch_reader.js:322]: more classes can be added as per need

**Comment by almchung** [comment]
Hi @aditya-shrivastavv, thank you for submitting this PR.
I shared my thoughts in the discussion thread (see #5753) as I feel that this approach may not completely resolve the underlying issue (of false positives and false negatives). If you would like to take a stab at a different approach, please feel free to do so — I can answer any questions you may have.
Also, regardless of the solution being used, please add unit tests to ensure that the OP’s issue would be resolved after the change and also to increase test coverage. For instance, you can mimic [this specific unit test](https://github.com/processing/p5.js/blob/70d58873e9ced7c9f82ec6fb9087f7338a097ae7/test/unit/core/error_helpers.js#L993-L1010) so that the OP’s case of declaring size variable before setup() would NOT throw an error after the changes.

File: src/core/friendly_errors/sketch_reader.js (hunk 1 above)
Comment (davepagurek) [reply src/core/friendly_errors/sketch_reader.js:322]: right, maybe something like `classesWithGlobalFunctions` would be more accurate?

**Comment by davepagurek** [comment]
Thanks for making this change @aditya-shrivastavv! Agreed with @almchung, let's copy that unit test they linked and pick a few example functions to check for (maybe a fill-related one, a typography-related one, a shape-related one?)

**Comment by aditya-shrivastavv** [comment]
Thanks, @davepagurek and @almchung for your feedback, I have read #5753 and understood what problems my approach can cause in different situations.
But, as suggested by @davepagurek in [this comment](https://github.com/processing/p5.js/issues/5753#issuecomment-1465256040), looking at our future goals, we might have to ignore some of the caveats. Since it's a matter of discussion whether we want to extract function names in the build process via regex or do it in the functions array way, what should be my next steps in this PR?
anyway I sincerely appreciate the input of @almchung on this problem

**Comment by aditya-shrivastavv** [comment]
I think there is a similar test already written for this, which detects the reassignment of the p5.js functions outside setup.
[here](https://github.com/processing/p5.js/blob/70d58873e9ced7c9f82ec6fb9087f7338a097ae7/test/unit/core/error_helpers.js#L963-L976)
is that what we want? should I write a new test or should I modify the existing one?

**Comment by almchung** [comment]
The previous tests didn't catch the false positive cases discussed in #5753 so we need new unit tests for those.

**Review by davepagurek — COMMENTED** [review]
Thanks for adding some tests! I think my only remaining comments are about naming things in the code, after that I think this is good to go!

**Review by davepagurek — APPROVED** [review]
Looks great, thanks @aditya-shrivastavv!

**Comment by aditya-shrivastavv** [comment]
hey, @davepagurek I think we should rename the identifier `globalFunctions` to something better to make it much more clear for future.

**Comment by davepagurek** [comment]
@aditya-shrivastavv if you want to make another PR renaming the variable that sounds good to me!

**Pull request merged**